Source/Core/ClippingPlaneCollection.js

```diff
@@ -364,22 +633,103 @@ define([
         return intersection;
     };
 
+    /**
+     * Returns a Number encapsulating the state for this ClippingPlaneCollection.
+     *
+     * Clipping mode is encoded in the sign of the number, which is just the plane count.
+     * Used for checking if shader regeneration is necessary.
+     *
+     * @returns {Number} A Number that describes the ClippingPlaneCollection's state.
+     * @private
+     */
+    ClippingPlaneCollection.prototype.clippingPlanesState = function() {
+        return this._unionClippingRegions ? this._planes.length : -this._planes.length;
+    };
+
+    /**
+     * Sets the owner for the input ClippingPlaneCollection if there wasn't another owner.
+     * Destroys the owner's previous ClippingPlaneCollection if setting is successful.
+     *
+     * @param {ClippingPlaneCollection} [clippingPlaneCollection] A ClippingPlaneCollection (or undefined) being attached to an object
+     * @param {Object} owner An Object that should receive the new ClippingPlaneCollection
+     * @param {String} key The Key for the Object to reference the ClippingPlaneCollection
+     * @private
+     */
+    ClippingPlaneCollection.setOwnership = function(clippingPlaneCollection, owner, key) {
+        // Don't destroy the ClippingPlaneCollection if it is already owned by newOwner
+        if (clippingPlaneCollection === owner[key]) {
+            return;
+        }
+        // Destroy the existing ClippingPlaneCollection, if any
+        owner[key] = owner[key] && owner[key].destroy();
+        if (defined(clippingPlaneCollection)) {
+            //>>includeStart('debug', pragmas.debug);
+            if (defined(clippingPlaneCollection._owner)) {
+                throw new DeveloperError('ClippingPlaneCollection should only be assigned to one object');
+            }
+            //>>includeEnd('debug');
+            clippingPlaneCollection._owner = owner;
+            owner[key] = clippingPlaneCollection;
+        }
+    };
+
     /**
      * Determines if rendering with clipping planes is supported.
      *
-     * @returns {Boolean} <code>true</code> if ClippingPlaneCollections are supported; otherwise, returns <code>false</code>
+     * @returns {Boolean} <code>true</code> if ClippingPlaneCollections are supported
+     * @deprecated
      */
     ClippingPlaneCollection.isSupported = function() {
-        return !FeatureDetection.isInternetExplorer();
+        return true;
```

Comment:
This should have a `deprecationWarning`

Reply:
@likangning93 did you add the deprecation warning here?